### DISCUSSION TITLE
How to generate delegates from an interface?

### DISCUSSION THREAD

**Discussion by kuuhakuDev** [AutoFixture/AutoFixture#1390] [Q&A] [answered]
public class UnitTest1
    {
        [Fact]
        public void PassingTest()
        {
            var fixture = new Fixture().Customize(new AutoMoqCustomization() { ConfigureMembers = true, GenerateDelegates = true });
            var sut = fixture.Create<Sut>();
            var list = sut.GetData<int>();
            Assert.NotEmpty(list);
        }
    }

    public interface IDataBase 
    {
        List<TData> GetData<TData>();
    }

    public class Sut
    {
        private readonly IDataBase _dataBase;
        public Sut(IDataBase database) => _dataBase = database;
        public List<TData> GetData<TData>()
        {
            return _dataBase.GetData<TData>();
        }
    }

When performing my test, `GetData` does not return any data. I know this can be fixed with `Moq.Setup` , but is there no implicit way to generate this data?

**Comment by Ergamon** [comment]
From what I tested, this seems like a bug while progressing in versions.

I am using NSubstitute and if I basically do a 1:1 copy of your code using AutoNSubstituteCustomization it works like you expected it.

I am quite sure the same thing should happen while using Moq and I guess it worked by the time the library was written.

**Comment by aivascu** [comment, marked as answer]
Hello @kuuhakuDev,
Apparently this scenario was never implemented in the Moq integration. Generic methods are [explicitly ignored in Moq](https://github.com/AutoFixture/AutoFixture/blob/master/Src/AutoMoqUnitTest/FixtureIntegrationWithAutoMoqCustomizationTest.cs#L366), however in other mocking libraries integrations the situation is the [opposite](https://github.com/AutoFixture/AutoFixture/blob/master/Src/AutoNSubstituteUnitTest/FixtureIntegrationTest.cs#L267).

Perhaps the developers that implemented in the feature in AutoMoq couldn't do it at the time or considered it harmful. Either way I'll have to try to implement it to see if there are any blockers.

**Reply by kuuhakuDev** [reply]
Thank you very much, I was searching the internet, and I was wasting time trying to do something that couldn't be done.
I hope that in future updates they will be able to implement it.
